### PR TITLE
Hausdorff loss

### PR DESCRIPTION
**Pull request by blackhat-coder**
issue [330](https://github.com/keras-team/keras-cv/issues/330)

Hausdorff distance is the maximum distance of a set to the nearest point in the other set.

Capture

Other variations of hausdorff distance include [weighted hausdorff distance](https://arxiv.org/pdf/1806.07564.pdf)

### CODE REVIEW

**Comment by tanzhenyu** [comment]
@blackhat-coder can you help us describe how this will be used? And more importantly, which paper uses this loss that improves model quality

**Comment by bhack** [comment]
@tanzhenyu  https://paperswithcode.com/paper/weighted-hausdorff-distance-a-loss-function

Also check https://paperswithcode.com/paper/weighted-hausdorff-distance-a-loss-function

File: keras_cv/losses/hausdorff_test.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+import numpy as np
+from keras_cv.losses.hausdorff import Hausdorff
+
+
+class HausdorffTest(tf.test.TestCase):
```
Comment (tanzhenyu) [inline keras_cv/losses/hausdorff_test.py:20]: can you add a test with a fully convolutional network that compiles with this loss?

**Review by tanzhenyu — CHANGES_REQUESTED** [review]

**Comment by LukeWood** [comment]
Thanks for the PR!  For now we're going to limit contributions to things on our critical path; it is not clear to me that this loss independently produces value for users.  We will definitely revisit this loss should a model architecture that makes use of this loss be introduced to the package.

Perhaps if this is a standalone loss that provides great value it could belong in core Keras; but we'd need to discuss what unique issue this loss solves.

Feel free to discuss on the issue thread if you would really like to see this loss introduced!

**Pull request closed without merging**